modules/fedimint-mint-client/src/lib.rs

```diff
@@ -533,6 +538,43 @@ impl MintClientInit {
             .await?;
         Ok(())
     }
+
+    /// Fetch epochs in a given range and send them over `sender`
+    ///
+    /// Since WASM's `spawn` does not support join handles, we indicate
+    /// errors via `sender` itself.
+    ///
+    /// TODO: could be internal to recovery_loop?
+    fn fetch_block_stream<'a>(
+        api: DynGlobalApi,
+        decoders: ModuleDecoderRegistry,
+        epoch_range: ops::Range<u64>,
+    ) -> impl futures::Stream<Item = (u64, SessionOutcome)> + 'a {
+        // How many request for blocks to run in parallel (streaming).
+        const PARALLISM_LEVEL: usize = 8;
+
+        futures::stream::iter(epoch_range)
+            .map(move |block_idx| {
+                let api = api.clone();
+                let decoders = decoders.clone();
+                Box::pin(async move {
+                    info!(block_idx, "Fetching epoch");
+
+                    let block = loop {
+                        info!(target: LOG_CLIENT_RECOVERY_MINT, block_idx, "Awaiting signed block");
+                        match api.await_block(block_idx, &decoders).await {
```

Comment:
Should we have retry behavior here? I guess not since we lose 7 already-fetched blocks in the worst case and hopefully have retry behavior further up?

Reply:
I don't understand. The end state is that there's one big stream for a given recovery, this is a loop fetching one particular block/session in order and retrying until it works. We don't loose anything.

Reply:
I was just blind